woocommerce/woocommerce
# CYS - AI flow: keep persisted query param when redirecting

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR keeps the `&ref=entrepreneur-signup` when redirecting. Also, it hides the arrow near the `Let's get creative label`.


I tried to keep the arrow redirecting the user to the "Business Description" page, but the fix isn't straightforward. Mostly related to: https://github.com/woocommerce/woocommerce/issues/46683


| Without the query param | With `&ref=entrepreneur-signup` query param  |
|--------|--------|
| ![image](https://github.com/woocommerce/woocommerce/assets/4463174/381a0787-a998-4b9a-b4b4-c36bcd4db7ab) | ![image](https://github.com/woocommerce/woocommerce/assets/4463174/e09838dd-0d6f-4da3-b938-70c9981a3a6c) | 



Closes #46821 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that you are on a WooCommerce site where the CYS AI flow works.
2. Visit `/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fdesign-with-ai&ref=entrepreneur-signup` 
3. Continue to flow.
4. Ensure that you are redirected to the WC Admin page by clicking on the logo on the top-left corner, and the `&ref=entrepreneur-signup` is still present in the URL.
5. Repeat steps 2-3.
6. Ensure that the no arrow is visible near the `Let's Creative Label`
7. Apply some changes.
8. Click save.
9. Click on "Back to Home".
10. Ensure that you are redirected to the WC Admin page and that the URL contains `&ref=entrepreneur-signup`.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

CYS - AI flow: keep persisted query param when redirecting.

#### Comment 


</details>
